metaflow/plugins/aip/aip.py

```diff
@@ -859,20 +863,15 @@ def _create_volume(
         )
         return (resource, volume)
 
-    def _set_container_labels(self, container_op: ContainerOp):
-        # TODO(talebz): A Metaflow plugin framework to customize tags, labels, etc.
-        container_op.add_pod_label("aip.zillowgroup.net/aip-pod-default", "true")
-
-        # https://github.com/argoproj/argo-workflows/issues/4525
-        # all argo workflows need istio-injection disabled, else the workflow hangs.
-        container_op.add_pod_label("sidecar.istio.io/inject", "false")
+    def _get_flow_labels(self) -> Dict[str, str]:
+        # function return variable
+        ret_flow_labels: Dict[str, str] = {}
 
         prefix = "metaflow.org"
-        container_op.add_pod_annotation(f"{prefix}/flow_name", self.name)
-        container_op.add_pod_annotation(f"{prefix}/step", container_op.name)
-        container_op.add_pod_annotation(f"{prefix}/run_id", METAFLOW_RUN_ID)
+        ret_flow_labels[f"{prefix}/flow_name"] = self.name
         if self.experiment:
-            container_op.add_pod_annotation(f"{prefix}/experiment", self.experiment)
+            ret_flow_labels[f"{prefix}/experiment"] = self.experiment
```

Comment:
As we already have experiments as flow labels, do you feel we can remove experiment from flow tags, or prevent it from being duplicated in flow labels? https://github.com/zillow/metaflow/blob/tz/AIP-7393-argo-labels/metaflow/plugins/aip/aip_cli.py#L296-L298

Right now in flow label you can see experiment twice:

>   labels:
>     **metaflow.org/experiment: e42**
>     metaflow.org/flow_name: HelloFlow
>     **metaflow.org/tag_e42: "true"**
>

Reply:
This is because the experiment name becomes a tag, see https://github.com/zillow/metaflow/blob/9d11eded9e453ebb2f0d816d40379fd064c164ce/metaflow/plugins/aip/aip_cli.py#L298-L297

and all tags become pod labels too :)

Reply:
Not going to block this PR on this minor label duplication, but experiment was a KFP concept, whereas we are not utilizing metaflow project or branches. Maybe we can find a better way to group flows while reducing the number of concepts.

Reply:
created https://zillowgroup.atlassian.net/browse/AIP-7769 which I'll tackle this sprint
thank you!